frontend/server/src/DAO/Problems.php

```diff
@@ -77,6 +77,65 @@ private static function addTagFilter(
         }
     }
 
+    /**
+     * @return list<int>
+     */
+    private static function getAccessibleAclIds(
+        ?int $identityId,
+        ?int $userId
+    ): array {
+        if (is_null($identityId) || is_null($userId)) {
+            return [];
+        }
+
+        $sql = '
+            SELECT DISTINCT acl_id
+            FROM User_Roles
+            WHERE user_id = ? AND role_id = ?
+
+            UNION
+
+            SELECT DISTINCT gr.acl_id
+            FROM Groups_Identities gi
+            INNER JOIN Group_Roles gr ON gr.group_id = gi.group_id
+            WHERE gi.identity_id = ? AND gr.role_id = ?
+        ';
+
+        /** @var list<array{acl_id: int}> */
+        $results = \OmegaUp\MySQLConnection::getInstance()->GetAll(
+            $sql,
+            [
+                $userId,
+                \OmegaUp\Authorization::ADMIN_ROLE,
+                $identityId,
+                \OmegaUp\Authorization::ADMIN_ROLE
+            ]
+        );
+
+        return array_map(fn($row) => intval($row['acl_id']), $results);
```

Comment:
Acá ya declaraste que `acl_id` es entero. Así que no es necesario el cast a entero

```suggestion
        return array_map(fn($row) => $row['acl_id'], $results);
```